x/cash/controller_test.go

```diff
@@ -254,3 +255,64 @@ func TestMoveCoins(t *testing.T) {
 		})
 	}
 }
+
+func TestBalance(t *testing.T) {
+	var helpers x.TestHelpers
+
+	newAddr := func() weave.Address {
+		_, a := helpers.MakeKey()
+		return a.Address()
+	}
+
+	store := store.MemStore()
+	ctrl := NewController(NewBucket())
+
+	addr1 := newAddr()
+	coin1 := x.NewCoin(1, 20, "BTC")
+	if err := ctrl.IssueCoins(store, addr1, coin1); err != nil {
+		t.Fatalf("cannot issue coins: %s", err)
+	}
+
+	addr2 := newAddr()
+	coin2_1 := x.NewCoin(3, 40, "ETH")
+	coin2_2 := x.NewCoin(5, 0, "DOGE")
+	if err := ctrl.IssueCoins(store, addr2, coin2_1); err != nil {
+		t.Fatalf("cannot issue coins: %s", err)
+	}
+	if err := ctrl.IssueCoins(store, addr2, coin2_2); err != nil {
+		t.Fatalf("cannot issue coins: %s", err)
+	}
+
+	cases := map[string]struct {
+		addr      weave.Address
+		wantCoins x.Coins
+		wantErr   error
+	}{
+		"non exising account": {
+			addr:    newAddr(),
+			wantErr: errors.NotFoundErr,
+		},
+		"exising account with one coin": {
+			addr:      addr1,
+			wantCoins: x.Coins{&coin1},
+		},
+		"exising account with two coins": {
+			addr:      addr2,
+			wantCoins: x.Coins{&coin2_1, &coin2_2},
```

Comment:
This is incorrect. We should expect all coins to be stored in normalized form, that is DOGE first

`wantCoins: x.Coins{&coin2_2, &coin2_1}, `

should be what you get

Reply:
I have rolled back the refactor in https://github.com/iov-one/weave/pull/316/commits/ab33529fd320f3bac324703b7c1e56aa6855bdc4